BUMP_VERSION.asciidoc

```diff
@@ -0,0 +1,196 @@
+is version 5, and Elasticsearch's current version of 5.x is 5.2.0.  The 
+next major release version is 6, which is reflected in the master branch.  
+
+== Steps to take for feature freeze
+
+When ready to feature freeze 5.2.0, the following steps must be taken.
+
+==== Create the branch
+
+Create a new `5.2` branch from the `5.x` branch and push it to Github.  
+The `5.2` branch will represent the 5.2 line of 5.x.
+-------------------------------------
+git checkout -b 5.2 upstream/5.x
+git push upstream 5.2
+-------------------------------------
+
+==== Update the version on the branch
+
+Update the version of the `5.x` branch to 5.3.0.  This requires the following
+steps:
+
+- Checkout a new branch for this work, from which you will issue a pull request against `5.x`: 
+-------------------------------------
+  `git checkout -b bump_version_to_5.3 upstream/5.x`
+-------------------------------------
+
+- In `core/src/main/java/org/elasticsearch/Version.java`, add a constant for V_5_3_0,
+mimicking how the constants are created for other versions.  Set the `CURRENT` constant
+to the newly created V_5_3_0 version constant.  Make sure the Lucene version is correct
+for the newly created version.
+-------------------------------------
+  + public static final int V_5_3_0_ID_UNRELEASED = 5030099;
+  + public static final Version V_5_3_0_UNRELEASED = new Version(V_5_3_0_ID, org.apache.lucene.util.Version.LUCENE_6_4_0);
+  + public static final Version CURRENT = V_5_3_0_UNRELEASED;
+-------------------------------------
+
+- Update the `Version#fromId` method to include the new 5.3.0 version:
+-------------------------------------
+   public static Version fromId(int id) {
+       switch (id) {
+  +       case V_5_3_0_ID_UNRELEASED:
+  +           return V_5_3_0_UNRELEASED;
+-------------------------------------
+
+- Update the elasticsearch version in `buildSrc/version.properties` to 5.3.0:
+-------------------------------------
+  -elasticsearch     = 5.2.0
+  +elasticsearch     = 5.3.0
+-------------------------------------
+
+- In `docs/groovy-api/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/java-api/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/java-rest/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/plugins/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/reference/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
```

Comment:
It might be better to give a command that can do these in one go:

From the root of the repository:

```bash
$ perl -p -i -e 's/(version:\s*)5.2.0/${1}5.3.0/g' `find docs -name index.asciidoc`
```

This has the advantage of not having to do it like an animal, and not needing to keep this list up to date if there are additions/subtractions.